packages/hoppscotch-common/src/newstore/settings.ts

```diff
@@ -146,6 +140,26 @@ const dispatchers = defineDispatchers({
 
     return result
   },
+  toggleNestedSetting(
+    currentState: SettingsDef,
+    {
+      settingKey,
+      property,
+    }: { settingKey: KeysMatching<SettingsDef, object>; property: string }
```

Comment:
```suggestion
    }: { settingKey: KeysMatching<SettingsDef, Record<string, boolean>>; property: string }
```

Also, can we have a well-defined type for the `property`?